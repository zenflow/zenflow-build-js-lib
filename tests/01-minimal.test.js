/* eslint-env jest */

const path = require('path')
const { flatten, repeat } = require('ramda')
const buildFixture = require('./util/buildFixture')
const { readFile, fileExists } = require('./util/fs')

describe('minimal', () => {
  let tempDir
  beforeAll(async () => {
    tempDir = await buildFixture('minimal', { extras: true })
  })
  test('snapshot (A) evaluated cjs', async () => {
    const file = path.join(tempDir, 'dist/minimal-fixture.cjs.js')
    expect(require(file)).toMatchSnapshot()
  })
  test('snapshot (A) evaluated umd', async () => {
    const file = path.join(tempDir, 'dist/minimal-fixture.umd.js')
    expect(require(file)).toMatchSnapshot()
  })
  test('snapshot (B) built cjs', async () => {
    const file = path.join(tempDir, 'dist/minimal-fixture.cjs.js')
    expect(await readFile(file)).toMatchSnapshot()
  })
  test('snapshot (B) built es', async () => {
    const file = path.join(tempDir, 'dist/minimal-fixture.es.js')
    expect(await readFile(file)).toMatchSnapshot()
  })
  test('snapshot (B) built umd', async () => {
    const file = path.join(tempDir, 'dist/minimal-fixture.umd.js')
    expect(await readFile(file)).toMatchSnapshot()
  })
  test('snapshot (C) built cjs.min', async () => {
    const file = path.join(tempDir, 'dist/minimal-fixture.cjs.min.js')
    expect(await readFile(file)).toMatchSnapshot()
  })
  test('snapshot (C) built es.min', async () => {
    const file = path.join(tempDir, 'dist/minimal-fixture.es.min.js')
    expect(await readFile(file)).toMatchSnapshot()
  })
  test('snapshot (C) built umd.min', async () => {
    const file = path.join(tempDir, 'dist/minimal-fixture.umd.min.js')
    expect(await readFile(file)).toMatchSnapshot()
  })
  test('source maps exist for each format', async () => {
    const files = flatten(
      ['cjs', 'es', 'umd'].map(format =>
        [false, true].map(minify =>
          path.join(
            tempDir,
            'dist',
            `minimal-fixture.${format}` + (minify ? '.min' : '') + '.js.map',
          ),
        ),
      ),
    )
    const filesExist = await Promise.all(files.map(fileExists))
    expect(filesExist).toEqual(repeat(true, 3 * 2))
  })
})
