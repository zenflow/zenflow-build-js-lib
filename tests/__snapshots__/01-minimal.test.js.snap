// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`minimal snapshot 1`] = `
Array [
  "/** @preserve
  * package: minimal-fixture v0.0.0
  * file: dist/minimal-fixture.cjs.js
  **/

'use strict';

var format = (words => words.join(' '));

const foo = 'foo';
const bar = 'bar';

var index = format([foo, bar]);

module.exports = index;
//# sourceMappingURL=minimal-fixture.cjs.js.map
",
  "/** @preserve
  * package: minimal-fixture v0.0.0
  * file: dist/minimal-fixture.cjs.min.js
  **/
\\"use strict\\";var format=o=>o.join(\\" \\");const foo=\\"foo\\",bar=\\"bar\\";var index=format([foo,bar]);module.exports=index;
//# sourceMappingURL=minimal-fixture.cjs.min.js.map
",
  "/** @preserve
  * package: minimal-fixture v0.0.0
  * file: dist/minimal-fixture.es.js
  **/

var format = (words => words.join(' '));

const foo = 'foo';
const bar = 'bar';

var index = format([foo, bar]);

export default index;
//# sourceMappingURL=minimal-fixture.es.js.map
",
  "/** @preserve
  * package: minimal-fixture v0.0.0
  * file: dist/minimal-fixture.es.min.js
  **/
var format=o=>o.join(\\" \\");const foo=\\"foo\\",bar=\\"bar\\";var index=format([foo,bar]);export default index;
//# sourceMappingURL=minimal-fixture.es.min.js.map
",
  "/** @preserve
  * package: minimal-fixture v0.0.0
  * file: dist/minimal-fixture.umd.js
  **/

(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? module.exports = factory() :
	typeof define === 'function' && define.amd ? define(factory) :
	(global.minimalFixture = factory());
}(this, (function () { 'use strict';

	var format = (words => words.join(' '));

	const foo = 'foo';
	const bar = 'bar';

	var index = format([foo, bar]);

	return index;

})));
//# sourceMappingURL=minimal-fixture.umd.js.map
",
  "/** @preserve
  * package: minimal-fixture v0.0.0
  * file: dist/minimal-fixture.umd.min.js
  **/
!function(e,n){\\"object\\"==typeof exports&&\\"undefined\\"!=typeof module?module.exports=n():\\"function\\"==typeof define&&define.amd?define(n):e.minimalFixture=n()}(this,function(){\\"use strict\\";return(e=>e.join(\\" \\"))([\\"foo\\",\\"bar\\"])});
//# sourceMappingURL=minimal-fixture.umd.min.js.map
",
]
`;
